### PR TITLE
Match string and number inputs to spec

### PR DESCRIPTION
**Pull request by rschamp**
_Preview:_
image

Also adds a "Say" block to demo string inputs.

I couldn't see a good way to override the looks of inputs just for horizontal blocks so this breaks the vertical style quite a bit.  Suggestions welcome...

### CODE REVIEW

File: blocks_horizontal/looks.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "id": "looks_say",
+      "message0": "%1 %2",
+      "args0": [
+        {
+          "type": "field_image",
+          "src": "../media/icons/looks_say.svg",
```
Comment (tmickel) [inline blocks_horizontal/looks.js]: let's make this relative as well!

Comment (rschamp) [inline blocks_horizontal/looks.js]: Oh, right — thanks!

File: core/field.js [hunk 2]
```diff
@@ -76,7 +78,7 @@ Blockly.Field.prototype.name = undefined;
  * Maximum characters of text to display before adding an ellipsis.
  * @type {number}
  */
-Blockly.Field.prototype.maxDisplayLength = 50;
+Blockly.Field.prototype.maxDisplayLength = 5;
```
Comment (tmickel) [inline core/field.js:81]: Someday we should think about where to put this. Should it be defined with the rest of the block?

File: core/field.js [hunk 3]
```diff
@@ -137,10 +139,14 @@ Blockly.Field.prototype.init = function(block) {
...
-       'height': 16}, this.fieldGroup_, this.sourceBlock_.workspace);
+       'height': Blockly.BlockSvg.FIELD_HEIGHT}, this.fieldGroup_, this.sourceBlock_.workspace);
   /** @type {!Element} */
   this.textElement_ = Blockly.createSvgElement('text',
-      {'class': 'blocklyText', 'y': this.size_.height - 12.5},
+      {'class': 'blocklyText',
+       'y': this.size_.height/2 + 6.25,
+       'x': Blockly.BlockSvg.FIELD_WIDTH / 2,
+       'width': Blockly.BlockSvg.FIELD_WIDTH - Blockly.BlockSvg.SEP_SPACE_X,
+       'text-anchor': 'middle'},
```
Comment (tmickel) [inline core/field.js:149]: and this section I'm guessing we'll have to reconsider for vertical

**Pull request merged**